CedarJava/src/main/java/com/cedarpolicy/model/PartialAuthorizationRequest.java

```diff
@@ -0,0 +1,100 @@
+package com.cedarpolicy.model;
+
+import com.cedarpolicy.model.schema.Schema;
+import com.cedarpolicy.model.slice.Entity;
+import com.cedarpolicy.value.EntityUID;
+import com.cedarpolicy.value.Value;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * A partial authorization request consists of an optional principal, action, and optional resource as well as an
+ * optional context mapping strings to Cedar values. When evaluating the request against a slice, the authorization
+ * engine determines if the policies allow for the given principal to perform the given action against the given
+ * resource. If a decision can be reached, then the response will provide the decision. If a decision can't be reached
+ * due to missing information Cedar will attempt to reduce the policies as much as possible and will return the residual
+ * policies.
+ *
+ * <p>If the (optional) schema is provided, this will inform parsing the
+ * `context` from JSON: for instance, it will allow `__entity` and `__extn`
+ * escapes to be implicit, and it will error if attributes have the wrong types
+ * (e.g., string instead of integer).
+ * If the schema is provided and `enable_request_validation` is true, then the
+ * schema will also be used for request validation.
+ */
+@JsonInclude(JsonInclude.Include.NON_ABSENT)
+public class PartialAuthorizationRequest extends AuthorizationRequest {
+    public PartialAuthorizationRequest(Optional<EntityUID> principalEUID, EntityUID actionEUID, Optional<EntityUID> resourceEUID, Optional<Map<String, Value>> context, Optional<Schema> schema, boolean enable_request_validation) {
```

Comment:
We don't enforce formatting in CI, but I think each of these args should go on separate lines.